### PR TITLE
Updated checkstyle to use local config

### LINKED ISSUE #1412: Enable checkstyle rule CommentsIndentation

### PR DESCRIPTION
**Pull request by cppwfs**
Checkstyle now checks for comments indentation

resolves #1412

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -212,6 +212,7 @@
 							<failOnViolation>true</failOnViolation>
 							<violationSeverity>warning</violationSeverity>
 							<includeTestSourceDirectory>true</includeTestSourceDirectory>
+							<configLocation>${checkstyle.config.location}</configLocation>
```
Comment (markpollack) [inline pom.xml]: So strange.  Look at the comment in line 202 above, not sure what the rationale is to keep `./spring-cloud-dataflow-server-local/pom.xml ` in sync w/ changes to this plugin.

Comment (cppwfs) [reply pom.xml]: Because in this case the Spring-cloud-dataflow-server-local uses the `spring-boot-starter-parent` as its parent.

**Comment by markpollack** [comment]
I get 
```[INFO] Starting audit...
[ERROR] /home/mpollack/projects/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TestDependencies.java:54: Wrong order for 'org.springframework.cloud.dataflow.server.controller.CompletionController' import. [ImportOrder]
[ERROR] /home/mpollack/projects/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TestDependencies.java:80: Wrong order for 'org.springframework.cloud.dataflow.server.service.SkipperStreamService' import. [ImportOrder]
[ERROR] /home/mpollack/projects/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TestDependencies.java:84: Wrong order for 'org.springframework.cloud.dataflow.server.service.impl.DefaultSkipperStreamService' import. [ImportOrder]
[ERROR] /home/mpollack/projects/spring-cloud-dataflow/spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSkipperStreamServiceIntegrationTests.java:47: Wrong order for 'org.springframework.cloud.dataflow.server.service.SkipperStreamService' import. [ImportOrder]
Audit done.
```

**Comment by cppwfs** [comment]
I just updated the PR.  Looks like some of the new files introduced since this PR was submitted had invalid import order.   Try now.

**Comment by cppwfs** [comment]
Should we close this PR until we resolve the IDE root issue?   Or should we default the settings to use only defaults for checkstyle and accept the changes as they stand?

**Comment by markpollack** [comment]
squashed and merged as babee3ee0af33a67819591d471d33875bf22ee8b

**Pull request closed without merging**